### ISSUE TITLE
How will it be verified that a formal company is involved, and not an independent person?

### ISSUE THREAD

**Issue by Triantafullenia-Doumani** [pondersource/peppol-python#5]
Depends on:

1) the country?
2) the continent?
3) global certificate?

**Comment by Triantafullenia-Doumani** [comment]
Αt present, only a few countries use peppol.

_"If there is no Peppol Authority in your country, you could sign the TIA with OpenPeppol as the Peppol Authority and re-contract
at a later date when an Authority is established nationally"_

More details here:
https://peppol.eu/wp-content/uploads/2021/02/How-to-set-up-a-Post-Award-Peppol-Access-Point_v1.6.pdf

**Comment by michielbdejong** [comment]
Yes, if I understand correctly, as an end-user (corner 1 or 4) you need to be registered as a legal entity in a participating country.
Then you can approach a gateway (corner 2 or 3) and open an account. This is not necessarily cheap; it may cost you 0 euros per month in some cases (for instance if you have a free account at moneybird.com, they will act as your Peppol gateway for free), or 1000 euros per month in some other cases. They will do a 'KYC' ('know your customer') process, checking that you really represent this legal entity. For instance when Ponder Source signed up at moneybird.com, they sent a letter in the mail to our front office in Amsterdam, and that letter contained a verification code to complete the registration.

When the gateway acts as corner 2 (sending gateway), they sign the SOAP message with a certificate that they received from the central Peppol authority when they got their license to operate as a sending gateway. The receiving gateway (corner 3) checks the signature of each incoming message, to see if it came from the gateway where the sender legal entity is registered.



One thing I'm thinking about is setting up a service that accepts both Peppol-signed invoices and self-signed invoices, but clearly showing the difference, a bit like a browser that can display both https:// and http:// websites. But in order to be true "Peppol", corners 1 and 4 need to be legal entities in some participating country, and corners 2 and 3 need to be licensed Peppol gateways with a valid certificate.
